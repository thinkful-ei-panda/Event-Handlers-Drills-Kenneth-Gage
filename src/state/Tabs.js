import React from 'react';

class Tabs extends React.Component {

    static defaultProps = { tabs: [] ,
                            content : ''   
                     };



    


    render() {
        const buttons = this.props.tabs.map((tab,i) =>
            <button key={i}>{tab.name}</button>
        )
        const currentTab =this.props.tabs[0];
        return (
            <div>
                {buttons}
                {
                    this.props.tabs.length ? 
                    <div className="content">
                        {currentTab.content}
                    </div> : '' 
                }
            </div>
        )
    }
}

export default Tabs;