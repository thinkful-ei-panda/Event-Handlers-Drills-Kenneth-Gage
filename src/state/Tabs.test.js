import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Tabs from './Tabs';

describe(`Tabs Component` , () => {
    it ('renders without errors', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Tabs />, div);
        ReactDOM.unmountComponentAtNode(div);
    })
    const tabsProp = [
        {
        name : 'Steve Jobs',
        content : 'why'
    },
    {
        name : 'died from ',
        content : 'do i '
    },
    {
        name : "ligma",
        content : 'have to do this so much '
    }
        ]
    it( 'renders the first tab by default', () => {
        const tree = renderer.create(<Tabs tabs={tabsProp}/>).toJSON( )
        expect(tree).toMatchSnapshot()
    })

})