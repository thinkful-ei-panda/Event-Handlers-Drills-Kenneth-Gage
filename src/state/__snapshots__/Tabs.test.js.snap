// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs Component renders the first tab by default 1`] = `
<div>
  <button>
    Steve Jobs
  </button>
  <button>
    died from 
  </button>
  <button>
    ligma
  </button>
  <div
    className="content"
  >
    why
  </div>
</div>
`;
